libs/resolver/lockfile.rs

```diff
@@ -343,15 +343,17 @@ impl<TSys: LockfileSys> LockfileLock<TSys> {
     let config = deno_lockfile::WorkspaceConfig {
       root: WorkspaceMemberConfig {
         package_json_deps: pkg_json_deps(root_folder.pkg_json.as_deref()),
-        dependencies: if let Some(map) = maybe_external_import_map {
-          deno_config::import_map::import_map_deps(map)
-        } else {
-          root_folder
-            .deno_json
-            .as_deref()
-            .map(|d| d.dependencies())
-            .unwrap_or_default()
-        },
+        dependencies: maybe_external_import_maps
+          .into_iter()
+          .flat_map(deno_config::import_map::import_map_deps)
+          .chain(
+            root_folder
+              .deno_json
+              .as_deref()
+              .map(|d| d.dependencies())
+              .unwrap_or_default(),
+          )
+          .collect(),
```

Comment:
Merged importmap approach won't work here either, since by spec the first importmap with resolved import takes precedence, yet in case of multiple jsr/npm imports - during the merge we might want to satisfy requirements of all import maps here, and this is only possible if we'll be able to access all the referenced import maps.